tests/args.vader

```diff
@@ -23,3 +23,18 @@ Execute (neomake#GetMakers):
   NeomakeTestsWaitForFinishedJobs
   AssertEqual map(copy(getloclist(0)), 'v:val.text'),
     \ ['1', '2', '3a 3b', '4', fname_abs]
+
+Execute (dict function given as maker args value):
+  let g:makerdict = {
+        \ 'exe': '/usr/bin/printf',
+        \ 'errorformat': '%f: %m',
+        \ 'append_file': 0,
+        \ 'mapexpr': 'expand("%:p") . ": " . v:val',
+        \ }
+  function g:makerdict.args(...) dict
+    return "'%d\\n' " . (self == g:makerdict)
+  endfunction
+  call neomake#Make(1, [makerdict])
+  NeomakeTestsWaitForFinishedJobs
+  AssertEqual getloclist(0).text, '1'
```

Comment:
>     (25/60) [EXECUTE] (X) Vim(call):E730: using List as a String

Should be `getloclist(0)[0].text`.
Or better: `AssertEqual map(copy(getloclist(0)), 'v:val.text'), ['1']`.